request_recorder.go

```diff
@@ -18,27 +19,38 @@ func newRequestRecorder() *requestRecorder {
 }
 
 type requestRecorder struct {
+	mtx               sync.RWMutex
 	acceptedRequests  []recordedRequest
 	unmatchedRequests []recordedRequest
 }
 
 func (r *requestRecorder) AcceptedRequests() []recordedRequest {
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
 	return r.acceptedRequests
```

Comment:
This is probably not enough. 
You should return a copy of the slice because the encapsulated slice might still change - new requests can be inserted if the slice has available capacity.

Reply:
True, fixed now by returning the copy of the slice.